gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiStateServiceImpl.java

```diff
@@ -270,6 +273,34 @@ public GenericApiEntity start(ExecutionContext executionContext, String apiId, S
         }
     }
 
+    @Override
+    public boolean startV2DynamicProperties(String apiId, EventManager eventManager) {
+        try {
+            log.debug("Start V2 Dynamic properties for API {}", apiId);
+            Api apiToUpdate = apiRepository.findById(apiId).orElseThrow(() -> new ApiNotFoundException(apiId));
+
+            eventManager.publishEvent(ApiEvent.START_DYNAMIC_PROPERTY_V2, apiToUpdate);
+            return true;
+        } catch (TechnicalException ex) {
+            log.error("An error occurs while trying to start V2 Dynamic properties for API {}", apiId, ex);
+            throw new TechnicalManagementException("An error occurs while trying to start V2 Dynamic properties for API " + apiId, ex);
+        }
+    }
+
+    @Override
+    public boolean startV4DynamicProperties(String apiId, EventManager eventManager) {
+        try {
+            log.debug("starting V4 Dynamic properties for API {}", apiId);
+            Api apiToUpdate = apiRepository.findById(apiId).orElseThrow(() -> new ApiNotFoundException(apiId));
+
+            eventManager.publishEvent(ApiEvent.START_DYNAMIC_PROPERTY_V4, apiToUpdate);
+            return true;
+        } catch (TechnicalException ex) {
+            log.error("An error occurs while trying to start V4 Dynamic properties for API {}", apiId, ex);
```

Comment:
do not log if we rethrow an exception https://gravitee.slab.com/posts/good-practices-for-handling-java-exceptions-and-logging-1dc30qlp